go/vt/topo/topoproto/shard.go

```diff
@@ -34,11 +36,16 @@ func ParseKeyspaceShard(param string) (string, string, error) {
 // and extracts the parts. If a shard is not specified, it's
 // returned as empty string.
 func ParseKeyspaceOptionalShard(keyspaceShard string) (string, string) {
-	last := strings.LastIndex(keyspaceShard, "/")
-	if last == -1 {
-		return keyspaceShard, ""
+	parts := parseKeyspaceShard(keyspaceShard)
+	if len(parts) != 2 {
+		return parts[0], ""
 	}
-	return keyspaceShard[:last], keyspaceShard[last+1:]
+	return parts[0], parts[1]
+}
+
+func parseKeyspaceShard(param string) []string {
+	rp := regexp.MustCompile("[:/]")
```

Comment:
regexp is probably overkill for this. Given that this will be a high QPS call, prefer using https://golang.org/pkg/strings/#LastIndexAny instead.